src/datadog/search/parser.rs

```diff
@@ -8,35 +10,44 @@ use super::{
 pub type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 
 /// Quick wrapper parse function to convert query strings into our AST
-pub fn parse(query: &str) -> Result<QueryNode, Error> {
-    // Clean up our query string
-    let clean_query = query.trim();
-    // If we have an empty query, we presume we're matching everything
-    if clean_query.is_empty() {
-        return Ok(QueryNode::MatchAllDocs);
-    }
-    // Otherwise parse and interpret the query
-    let mut ast = EventPlatformQuery::parse(super::grammar::Rule::queryroot, query)?;
-    let rootquery = ast.next().ok_or("Unable to find root query")?;
-    let q = QueryVisitor::visit_queryroot(rootquery, DEFAULT_FIELD);
-    Ok(q)
+impl FromStr for QueryNode {
+    type Err = Error;
+
+    fn from_str(query: &str) -> Result<Self, Self::Err> {
+        // Clean up our query string
+        let clean_query = query.trim();
+        // If we have an empty query, we presume we're matching everything
+        Ok(if clean_query.is_empty() {
+            Self::MatchAllDocs
+        } else {
+            // Otherwise parse and interpret the query
+            let mut ast = EventPlatformQuery::parse(super::grammar::Rule::queryroot, query)?;
+            let rootquery = ast.next().ok_or("Unable to find root query")?;
+            QueryVisitor::visit_queryroot(rootquery, DEFAULT_FIELD)
+        })
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::super::node::{BooleanType, Comparison, ComparisonValue, QueryNode};
     use super::*;
 
+    fn parse(s: &str) -> QueryNode {
+        s.parse()
+            .unwrap_or_else(|_| panic!("Unable to parse {s:?}."))
```

Comment:
I think this is more idiomatic?
```suggestion
            .expect("Unable to parse {s:?}.")
```